### PR TITLE
[GPU] primitive serialization

### PR DESCRIPTION
**Pull request by e-ddykim**
### Details:
 - This PR enables serialization of all primitives.

### Tickets:
 - 110986

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/graph/serialization/activation_serializer.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2018-2023 Intel Corporation
```
Comment (andrew-k-park) [inline src/plugins/intel_gpu/include/intel_gpu/graph/serialization/activation_serializer.hpp:1]: nit: copyright notices for the new source code
```suggestion
// Copyright (C) 2023 Intel Corporation
```

File: src/plugins/intel_gpu/include/intel_gpu/graph/serialization/tensor_serializer.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2018-2023 Intel Corporation
```
Comment (andrew-k-park) [inline src/plugins/intel_gpu/include/intel_gpu/graph/serialization/tensor_serializer.hpp:1]: nit: same as above

**Review by andrew-k-park — APPROVED** [review]

File: src/plugins/intel_gpu/include/intel_gpu/graph/serialization/activation_serializer.hpp (hunk 1 above)
Comment (e-ddykim) [reply src/plugins/intel_gpu/include/intel_gpu/graph/serialization/activation_serializer.hpp]: I updated the copyright year. Thank you.

File: src/plugins/intel_gpu/include/intel_gpu/graph/serialization/tensor_serializer.hpp (hunk 2 above)
Comment (e-ddykim) [reply src/plugins/intel_gpu/include/intel_gpu/graph/serialization/tensor_serializer.hpp]: I updated the copyright year here too. Thank you.

**Review by ahnyoung-paul — APPROVED** [review]

**Comment by yeonbok** [comment]
As far as I understand, the primitive serilization is only needed for dynamic shape. Then, can we do it only for dynamic program to reduce the cached bin size for static model?

**Comment by e-ddykim** [comment]
> As far as I understand, the primitive serilization is only needed for dynamic shape. Then, can we do it only for dynamic program to reduce the cached bin size for static model?

Some primitives (such as quantize) are also required for static models too. And the increased blob size is not much. So, if possible, I'd like to use the same logic for static and dynamic models.

**Pull request merged**